python/TestHarness/schedulers/QueueManager.py

```diff
@@ -0,0 +1,403 @@
+               os.path.splitext(file)[1] != '':
+                shutil.copy('../' + file, '.')
+
+        # Copy directories
+        if 'copy_files' in template.keys():
+            for file in template['copy_files']:
+                if os.path.isfile('../' + file):
+                    if not os.path.exists(file):
+                        shutil.copy('../' + file, '.')
+                elif os.path.isdir('../' + file):
+                    try:
+                        shutil.copytree('../' + file, file)
+                    except OSError, ex:
+                        if ex.errno == errno.EEXIST: pass
+                        else: raise Exception()
+
+        # Create symlinks (do last, to allow hard copy above to be default)
+        if tester.specs.isValid('link_files'):
+            for file in tester.specs['link_files']:
+                if os.path.exists('../' + file):
+                    try:
+                        os.symlink('../' + file, file)
+                    except OSError, ex:
+                        if ex.errno == errno.EEXIST: pass
+                        else: raise Exception()
+
+        # return to the previous directory
+        os.chdir(current_pwd)
+
+    def downstreamFinished(self, job_container):
+        """ Return bool for given job_container for all downstream jobs finished """
+        job_dag = job_container.getOriginalDAG()
+        downstream_jobs = job_dag.all_downstreams(job_container)
+        for job in downstream_jobs:
+            test_unique = self.getUnique(job)
+            json_job = self.getData(test_unique, isFinished=True)
+
+            # None would mean the upstream job never launched
+            if 'isFinished' not in json_job.keys() or json_job['isFinished'] is False:
+                return False
+        return True
+
+    def queueJobs(self, status_jobs=[], run_jobs=[]):
+        """
+        Entry point into the QueueManager. Supply queueJobs with a list
+        of jobs to either launch, or print statuses for.
+
+        Syntax:
+
+           Launch job:
+            queueJobs(run_jobs=[list of jobs])
+
+           Print status:
+            queueJobs(status_jobs=[list of jobs])
+        """
+
+        # if __status_check is True, we do not care which 'queue' might have tried to initiate.
+        # QueueManager knows best when this is True.
+        if self.__status_check:
+            # Collect all jobs no matter the queue other schedulers are attempting to use.
+            # __status_check is True, so QueueManager knows best ATM.
+            all_jobs = set(status_jobs).union(run_jobs)
+
+            launched_dags = []
+            for job_container in all_jobs:
+                test_unique = self.getUnique(job_container)
+
+                # Only interested in jobs contained in the current session
+                if test_unique not in self.__session_data.keys():
+                    continue
+
+                # Yo dawg, I heard you like DAGs
```

Comment:
This isn't the right usage for this meme. "Yo dawg" is all about nesting things inside of things. You just have a lot at one level!